### PR TITLE
[delete] surplus code branches in kservice.c

### PR DESCRIPTION
**Pull request by ErnestChen1**
## 拉取/合并请求描述：(PR description)

[
  Delete surplus code branches in kservice.c, using kservice_tc which is about a large number testing rt_vsnprintf.

  test on stm32f407 and qemu with testcase about kservice，run well.
]

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other style
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality

### CODE REVIEW

**Comment by ErnestChen1** [comment]
@SummerLife

**Comment by ErnestChen1** [comment]
> 请对本次修改作出更加详细的描述。

多余重复分支，删除

File: src/kservice.c [hunk 1]
```diff
@@ -999,21 +999,17 @@ rt_int32_t rt_vsnprintf(char       *buf,
...
-            if (flags & SIGN) num = (rt_int32_t)num;
-        }
-        else if (qualifier == 'h')
         {
             num = (rt_uint16_t)va_arg(args, rt_int32_t);
             if (flags & SIGN) num = (rt_int16_t)num;
         }
         else
         {
+            /* include qualifier == 'l', executing the same program */
```
Comment (Guozhanxin) [inline src/kservice.c:1012]: L l h 分别代表 long long\long\short 三种类型数据。这里本来分别判断了三种类型并做了分别的处理，然后，针对这三种之外的类型，按照 l(long) 型进行处理。但是代码检查工具看起来，l 的处理分支和其他类型的处理分支是重复的。

Comment (ErnestChen1) [reply src/kservice.c:1012]: 实际也是重复，多余的判断类型而已

**Pull request closed without merging**